src/Extension/FluentFilteredExtension.php

```diff
@@ -148,4 +154,22 @@ protected function getDataQueryLocale(DataQuery $dataQuery = null)
 
         return null;
     }
+
+    /**
+     * There are two different DRAFT modes. One when browsing stage, and one when browsing archive. Both modes have
+     * "Stage" at the very end of their reading_mode name.
+     *
+     * @return bool
+     */
+    private function getModeIsStage()
+    {
+        $str = Versioned::get_reading_mode();
+        $test = Versioned::DRAFT;
+
+        if (strlen($str) === 0) {
+            $str = Versioned::DEFAULT_MODE;
+        }
+
+        return substr_compare($str, $test, strlen($str) - strlen($test), strlen($test)) === 0;
```

Comment:
Is there a reason we can't use `Versioned::get_stage()` or a combination of that and `Versioned::current_archived_stage()`? I'd really like to try and use Versioned API methods here if possible

Reply:
I did this because `current_archived_stage` returns a value of `Stage` if you are browsing the `Live` site.

![screen shot 2018-09-07 at 6 57 34 am](https://user-images.githubusercontent.com/505788/45179043-7a6cde00-b26b-11e8-98f1-cc19564a2b3f.png)

`get_reading_mode` is consistent though - you get `stage.Stage` or `Archive.[date].Stage`. This seemed like the most succinct/consistent way to check if I'm browsing in `Stage` in or outside of Archive.